### ISSUE TITLE
A Strange Behavior in TensorFlow about Conv2d Creation and Weight Optimization

### ISSUE THREAD

**Issue by mabbs90** [tensorflow/tensorflow#8506]
To simplify the tensorflow syntax, I created utility functions that would generate a conv2d layer more easily for me. However, when I use it in an architecture, the results are bizzare and different from the vanilla coding version. Even though both methods are returning the same tensor shape, tensorflow fails to optimize the weights in the second approach and I always get extremely low accuracies, vs the first version returns a perfect accuracy.

It seems like every time Tensorflow initializes the weights randomly instead of optimizing it. Is this a known bug?

Version 1
---
`sigma = 0.1`
`x = tf.placeholder(tf.float32, (None, 32, 32, 3))`

`conv1_W = tf.Variable(tf.truncated_normal(shape=(5, 5, 3, 6), mean = mu, stddev = sigma))`
`conv1_b = tf.Variable(tf.zeros(6))`
`conv1 = tf.nn.conv2d(x, conv1_W, strides=[1, 1, 1, 1], padding='VALID') + conv1_b`

Version 2
---

`def weights(dims, mu = 0, sigma = 0.1):`
&nbsp;&nbsp;&nbsp;&nbsp;`w = tf.Variable(tf.truncated_normal(shape = dims, mean = mu, stddev = sigma))`
&nbsp;&nbsp;&nbsp;&nbsp;`return w`

`def bias(dims):`
&nbsp;&nbsp;&nbsp;&nbsp;`b = tf.Variable(tf.zeros(dims))`
&nbsp;&nbsp;&nbsp;&nbsp;`return b`

`def conv(layer, dims, stride = [1, 1, 1, 1], pad = 'VALID'):`
&nbsp;&nbsp;&nbsp;&nbsp;`w = weights(dims)`
&nbsp;&nbsp;&nbsp;&nbsp;`b = bias(dims[-1])`
&nbsp;&nbsp;&nbsp;&nbsp;`conv2 = tf.nn.conv2d(layer, w, strides = stride, padding = pad) + b`
&nbsp;&nbsp;&nbsp;&nbsp;`return conv2`
this line is the only line that I put in my network architecture: so I created my conv2d layers like this:

`conv2 = conv(x, [5, 5, 3, 6]) `

**Comment by concretevitamin** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by mabbs90** [comment]
I don't see why it's not a bug! otherwise I would've posted in on stackoverflow. Anyways, thanks!
